addons/stock/product.py

```diff
@@ -84,7 +85,15 @@ def _get_domain_locations(self, cr, uid, ids, context=None):
                 location_ids = context['location']
         else:
             if context.get('warehouse', False):
-                wids = [context['warehouse']]
+                if type(context['warehouse']) == type(1):
+                    wids = [context['warehouse']]
+                elif type(context['warehouse']) in (type(''), type(u'')):
+                    domain = [('name', 'ilike', context['warehouse'])]
+                    if context.get('force_company', False):
+                        domain += [('company_id', '=', context['force_company'])]
+                    wids = warehouse_obj.search(cr, uid, domain, context=context)
+                else:
+                    wids = context['warehouse']
```

Comment:
It is not better to change *<field name="location_id" widget="selection"
context="{'location': self}"/>*
*                    <field name="warehouse_id" widget="selection"
context="{'warehouse': self}"/> *

*in product_search_form_view_stock?  Is that the reason you want to do
this? (could we do self.name <http://self.name> or self.id
<http://self.id>)*

On Thu, Feb 19, 2015 at 11:37 AM, Jesús Ventosinos Mayor <
notifications@github.com> wrote:

> In addons/stock/product.py
> <https://github.com/odoo/odoo/pull/5218#discussion_r24978601>:
>
> > @@ -84,7 +85,15 @@ def _get_domain_locations(self, cr, uid, ids, context=None):
> >                  location_ids = context['location']
> >          else:
> >              if context.get('warehouse', False):
> > -                wids = [context['warehouse']]
> > +                if type(context['warehouse']) == type(1):
> > +                    wids = [context['warehouse']]
> > +                elif type(context['warehouse']) in (type(''), type(u'')):
> > +                    domain = [('name', 'ilike', context['warehouse'])]
> > +                    if context.get('force_company', False):
> > +                        domain += [('company_id', '=', context['force_company'])]
> > +                    wids = warehouse_obj.search(cr, uid, domain, context=context)
> > +                else:
> > +                    wids = context['warehouse']
>
> The checking of type integer or string is necessary. The else for lists
> can be deleted, but does not change much the code.
> The force_company checking it could also be removed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/odoo/odoo/pull/5218/files#r24978601>.
>